src/java/htsjdk/samtools/CRAMIterator.java

```diff
@@ -83,10 +83,10 @@ public void setValidationStringency(
     public CRAMIterator(final InputStream inputStream, final ReferenceSource referenceSource)
             throws IOException {
         if (null == referenceSource) {
-            throw new CRAMException("A reference source is required for CRAM files");
-        }
+            this.referenceSource = new ReferenceSource();
+        } else
+            this.referenceSource = referenceSource;
```

Comment:
Add curly braces around else block. Also, falling back to a default-constructed `ReferenceSource` will (I believe) eventually produce a `NullPointerException` deep in the cram parsing code provided that `Defaults.USE_CRAM_REF_DOWNLOAD` is false (which it is by default). I think we should throw here if `referenceSource` is null and `Defaults.USE_CRAM_REF_DOWNLOAD` is false.